### ISSUE TITLE
[Feature]: Support the `step` option for the `Locator.dragTo` method

### ISSUE THREAD

**Issue by romanresh** [microsoft/playwright#32609]
### 🚀 Feature Request

By default, the Locator.dragTo method raises one intermediate mousemove event. 
We need to set the `steps` option to increase number of intermediate mousemove events as it's implemented in the [Mouse.Move method options](https://playwright.dev/docs/api/class-mouse#mouse-move-option-steps).

### Example

Set the non-default `steps` option:

`await source.dragTo(target, {
  steps: 3
});`

### Motivation

There are a few cases:
1. In some components, dragging starts only after a few intermediate mousemove events with holding mousedown to prevent unintended changes.
2. Sometimes, during the dragging action some elements should catch dragging above them before dragging is finished on the end point.

**Comment by dgozman** [comment]
@romanresh While we consider this feature request for prioritization, you can probably work around it by calling `page.mouse.down()/move()/up()` and passing steps there.

**Comment by romanresh** [comment]
> @romanresh While we consider this feature request for prioritization, you can probably work around it by calling `page.mouse.down()/move()/up()` and passing steps there.

Yes, thank you, we use this WA in our tests. 

It makes tests less stable because the drag event also makes a few checks before performing drag and drop: stabilize element, visibility and others. So, it's also important to call the `Locator.dragTo(.., { trial: true })` before `down/move/up`.

All these actions allow to test our scenarios, but the result trace log contains too much excess information.

**Comment by NabilNYMansour** [comment]
Ideally, it would also be better to have the ability to wait while dragging as some issues I've faced require a bit of waiting to be resolved. Something like this
```js
  await page.mouse.move(sourceBox.x + sourceBox.width / 2, sourceBox.y + sourceBox.height / 2);
  await page.mouse.down();
  await page.waitForTimeout(2000);
  await page.mouse.move(targetBox.x + targetBox.width / 2, targetBox.y + targetBox.height / 2, { steps: 5 });
  await page.waitForTimeout(2000);
  await page.mouse.up();
```

It would be nicer if that can be an option inside the `dragTo` function

**Comment by itsdouges** [comment]
@NabilNYMansour I've had similar issues where the mouse events would be triggered out of order. I ended up dispatching events to get the expected order.

Have you seen another issue related to this?

**Comment by Harmageddon** [comment]
Playwright 1.57 supports the `steps` parameter on `locator.dragTo`: https://playwright.dev/docs/api/class-locator#locator-drag-to

**Comment by dgozman** [comment]
@Harmageddon Indeed! This was implemented in #38184.
